### PR TITLE
Docker for Development

### PR DESCRIPTION
**Pull request by mlocher**
Use Docker for building the Documentation pages locally as well.

This will require everybody who wants to test the documentation locally to have Docker available but will bring the development environment more in line with CI and how we build and deploy the site.

It will also isolate changes to the Jekyll environment from unrelated changes on a developers machine.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,27 +1,56 @@
...
 
+[![Codeship Status for codeship/documentation](https://codeship.com/projects/0bdb0440-3af5-0133-00ea-0ebda3a33bf6/status?branch=master)](https://codeship.com/projects/102044)
 [![Waffle.io Board](https://badge.waffle.io/codeship/documentation.svg?label=ready&title=Ready)](http://waffle.io/codeship/documentation)
 [![Dependency Status](https://gemnasium.com/codeship/documentation.svg)](https://gemnasium.com/codeship/documentation)
 [![License](http://img.shields.io/:license-mit-blue.svg)](https://github.com/codeship/documentation/blob/master/LICENSE.md)
 
 ## Contributing
 
-See our [contributing guidelines](CONTRIBUTING.md) and our [code of conduct](CODE_OF_CONDUCT.md)
+We are happy to get feedback on our documentation or accept PRs from you. If you submit a PR or open a ticket, please read our [contributing guidelines](CONTRIBUTING.md) and the [code of conduct](CODE_OF_CONDUCT.md) first.
```
Comment (BoVice) [inline README.md]: I would revise to something a little less redundant like the following:

> 
We are happy to hear your feedback! Please read our [code of conduct](CODE_OF_CONDUCT.md) and [contributing guidelines](CONTRIBUTING.md) on how to contribute to Codeship's documentation.

File: README.md [hunk 2]
```diff
@@ -1,27 +1,56 @@
...
 
 ## Getting Started
 
 ### Setup
 
-Clone the repository, install needed gems and run Jekyll ;)
+To work on the project, you need to clone the repository
```
Comment (BoVice) [inline README.md]: Would make this more terse, for example:

> Clone this repository to your local dev machine:
>  ```bash
>   git clone git@github.com:codeship/documentation.git
>   cd documentation
> ```

File: README.md [hunk 3]
```diff
@@ -1,27 +1,56 @@
...
 
 ```bash
 git clone git@github.com:codeship/documentation.git
 cd documentation
-bundle install
-bundle exec jekyll serve
 ```
 
-You can then access the site at [http://localhost:4000](http://localhost:4000)
+If you have Docker installed, we'd recommend using Docker for running & building the documentation.
```
Comment (BoVice) [inline README.md]: If we are changing our documentation build environment over to Docker, but docker is not required, please provide steps for non-docker folks. Maybe write something like:

> We use Docker for building and running our Documentation page and recommend that our contributors use Docker as well. If you do not user docker, you can still run your builds by 
> - blah
> - blah
> - blah

**Comment by BoVice** [comment]
@mlocher Let's have the README.md follow the https://github.com/codeship/documentation/blob/master/templates/documentation-article.md. Let's also add the documentation template to the Contributing subheaded. 

While on the subject, why did we rename and move the documentation article to a "template" folder?

File: README.md (hunk 3 above)
Comment (mlocher) [inline README.md]: We don't see external contributions at the moment and for most contributions people don't actually need to build the site (e.g. when they fix a typo), so I omitted that information on purpose for now.

I have a couple improvements on the roadmap which will make building the Docker image unnecessary for most users and then write wrapper scripts for the most common commands which check for Docker support and fall back to running Jekyll locally if it's not found.

**Comment by mlocher** [comment]
We're basically following the tutorial already, from what I can tell, what's missing is the prerequisite section and the conclusion. A conclusion imo doesn't make sense for the Readme. I'll add the prerequisites with my next update.

As for moving the tutorial to a subfolder: because of #394, which you reviewed. For a bit more information, because it was accessible on the hosted documentation and I expect more templates to follow and having a subfolder that is ignored by Jekyll is easier than having to update the list each time.

To make it even more clear, that the folder isn't part of the documentation I should have named it `_templates`, but we can do this in a later PR as well.

File: README.md [hunk 4]
```diff
@@ -1,27 +1,62 @@
...
 
+[![Codeship Status for codeship/documentation](https://codeship.com/projects/0bdb0440-3af5-0133-00ea-0ebda3a33bf6/status?branch=master)](https://codeship.com/projects/102044)
 [![Waffle.io Board](https://badge.waffle.io/codeship/documentation.svg?label=ready&title=Ready)](http://waffle.io/codeship/documentation)
 [![Dependency Status](https://gemnasium.com/codeship/documentation.svg)](https://gemnasium.com/codeship/documentation)
 [![License](http://img.shields.io/:license-mit-blue.svg)](https://github.com/codeship/documentation/blob/master/LICENSE.md)
 
 ## Contributing
 
-See our [contributing guidelines](CONTRIBUTING.md) and our [code of conduct](CODE_OF_CONDUCT.md)
+We are happy to hear your feedback. Please read our [contributing guidelines](CONTRIBUTING.md) and the [code of conduct](CODE_OF_CONDUCT.md) before you submit a PR or open a ticket.
```
Comment (BoVice) [inline README.md]: Instead of PR => Pull Request

File: README.md [hunk 5]
```diff
@@ -1,27 +1,62 @@
...
 
 ## Getting Started
 
+## Prerequisites
+
+* [Docker](https://docs.docker.com/engine/installation/)
+
+We recommend using Docker to build and test the documentation. Running via Docker is only required if you plan to make changes to the styling or layout of the site.
+
+For content related changes and fixes, it's easiest to use GitHubs [File Edit UI](https://help.github.com/articles/editing-files-in-another-user-s-repository/) to make the changes and create the PR.
```
Comment (BoVice) [inline README.md]: PR => Pull Request

File: README.md [hunk 6]
```diff
@@ -1,27 +1,62 @@
...
+
 ### Setup
 
-Clone the repository, install needed gems and run Jekyll ;)
+To work on the project, you need to clone the repository
```
Comment (BoVice) [inline README.md]: Can we remove "To work on the project, you need to". This is redundant and a bit too wordy

File: README.md [hunk 7]
```diff
@@ -1,27 +1,62 @@
...
+docker run -it -p 4000:4000 -v $(pwd):/docs codeship/documentation
+```
+
+To access the site open http://IP_OF_YOUR_DOCKER_SERVER:4000 in your browser.
+
+## Development
+
+### Updating dependencies
+
+To update Rubygem based dependencies make sure to update the `Gemfile` (if required) and run
```
Comment (BoVice) [inline README.md]: Add comma after "dependencies,"

Remove "make sure"

File: README.md [hunk 8]
```diff
@@ -1,27 +1,62 @@
...
+
+```bash
+docker run -it -v $(pwd):/docs codeship/documentation bundle update
+```
+
+For NPM based dependencies run the following two commands
```
Comment (BoVice) [inline README.md]: comma after "dependencies"

File: README.md [hunk 9]
```diff
@@ -64,7 +111,7 @@ generate the output like the following (depending on configuration values)
 
 #### Man Pages
 
-Link to a specific Ubuntu man page. This currently defaults to the Ubuntu Trusty version.
+Link to a particular Ubuntu man page. The Ubuntu version currently defaults to Ubuntu Trusty.
```
Comment (BoVice) [inline README.md]: I'm a little confused with this sentence. Would it be more appropriate to make a "NOTE:" of the default OS version?

**Comment by BoVice** [comment]
Added some more comments to minor things. Besides that, :shipit:

File: README.md (hunk 9 above)
Comment (mlocher) [inline README.md]: That sentence has been in the Readme since you started working for the company. 

The function creates a link to a manpage on the Ubuntu website, e.g. `{% man_url nohup %}` would expand to `http://manpages.ubuntu.com/manpages/trusty/en/man1/nohup.1.html`. That URL includes a OS version (in this case `trusty`). The OS version is configurable via the `_config.yml` file, and is currently set to `trusty`, which is noted in the Readme. 

Full source for the helper is available in [man_tag.rb](https://github.com/codeship/documentation/blob/master/_plugins/man_tag.rb)

Comment (BoVice) [inline README.md]: I wasn't questioning the content, just the wording. but doesn't matter, you already merged it.

Comment (mlocher) [inline README.md]: I just made the changes suggested by Grammarly, didn't touch that section otherwise, but we can always make it clearer in a different PR. 

If you think this is unclear, please prepare such a PR and we will take a look.
